code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,262 @@
+
+/// Mail is tamper-evident and unresealable, postmarked by CentCom for an individual recepient.
+/obj/item/mail
+	name = "mail"
+	gender = NEUTER
+	desc = "An officially postmarked, tamper-evident parcel regulated by CentCom and made of high-quality materials."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mail_small"
+	inhand_icon_state = "paper"
+	worn_icon_state = "paper"
+	item_flags = NOBLUDGEON
+	w_class = WEIGHT_CLASS_SMALL
+	drop_sound = 'sound/items/handling/paper_drop.ogg'
+	pickup_sound =  'sound/items/handling/paper_pickup.ogg'
+	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
+	/// destination tagging for the mail sorter
+	var/sortTag = 0
+	/// who this mail is for and who can open it
+	var/mob/recipient
+	/// how many goodies this mail contains
+	var/goodie_count = 1
+	/// goodies which can be given to anyone.
+	/// the base weight for cash is 56. for there to be a 50/50 chance of getting a department item, they need 56 weight as well.
+	var/list/generic_goodies = list(
+		/obj/item/stack/spacecash/c100 = 30,
+		/obj/item/stack/spacecash/c200 = 20,
+		/obj/item/stack/spacecash/c500 = 10,
+		/obj/item/stack/spacecash/c1000 = 5,
+		/obj/item/stack/spacecash/c10000 = 1
+	)
+	// Overlays (pure fluff)
+	var/postmarked = TRUE
+	var/stamped = TRUE
+	var/list/stamps = list()
+	var/stamp_max = 1
+	var/stamp_offset_x = 0
+	var/stamp_offset_y = 2
+
+	///mail will have the color of the department the recipient is in.
+	var/static/list/department_colors
+
+/obj/item/mail/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	goodie_count = 2
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/Initialize()
+	. = ..()
+
+	department_colors = list(
+	ACCOUNT_CIV = COLOR_WHITE,
+	ACCOUNT_ENG = COLOR_PALE_ORANGE,
+	ACCOUNT_SCI = COLOR_PALE_PURPLE_GRAY,
+	ACCOUNT_MED = COLOR_PALE_BLUE_GRAY,
+	ACCOUNT_SRV = COLOR_PALE_GREEN_GRAY,
+	ACCOUNT_CAR = COLOR_BEIGE,
+	ACCOUNT_SEC = COLOR_PALE_RED_GRAY,
+	)
+
+	// Icons
+	// Add some random stamps.
+	if(stamped == TRUE)
+		var/N = rand(1, stamp_max)
+		for(var/i = 1, i <= N, i++)
```

Comment:
for(var/i to num)

Reply:
This shorthand causes the compiler to throw warnings, because the variables are used indirectly with the loop.
```
code\game\objects\items\mail.dm:62:warning: num: variable defined but not used
code\game\objects\items\mail.dm:63:warning: i: variable defined but not used
```

Reply:
its for(var/i in 1 to num)